docs/using.md

```diff
@@ -183,6 +183,46 @@ See [the example PipelineRun](../examples/invocations/kritis-pipeline-run.yaml).
    that the `Task` needs to run.
 2. The `TaskRun` will also serve as a record of the history of the invocations of the
    `Task`.
+3. Another way of running a Task is embedding the TaskSpec in the taskRun yaml as shown in the following example
+
+```yaml
+apiVersion: pipeline.knative.dev/v1alpha1
```

Comment:
So we still need to create two resources to run a simple set of steps? How can we drop the requirement that a PipelineResource already exists?

In Build, resources were specified directly in the build itself, as `source(s)`.

Reply:
I think adding even more embedding is just making the yaml file too complicated. Also one side benefit is that the resources are such a core concept in build-pipeline that we want users to know about when migrating.

Maybe as part of the work for the resources extensibility we can make this easier by making the taskRun controller create the resource for for system supported resource types such as git. In that case, the we can define types for system resources with their specific fields to simplify yaml files for all tasks

Depending on what we decide here, how big the change is, it might be better to have another follow up PR so this one doesn't get too big

Reply:
> I think adding even more embedding is just making the yaml file too complicated. Also one side benefit is that the resources are such a core concept in build-pipeline that we want users to know about when migrating.

That's a good point!

> Depending on what we decide here, how big the change is, it might be better to have another follow up PR so this one doesn't get too big

Sounds good to me